File: src/1041-Robot-Bounded-In-Circle/1041.py
class Solution:
    def isRobotBounded(self, instructions: str) -> bool:
        x, y, dx, dy = 0, 0, 0, 1
        for i in instructions:
            if i == 'R': 
                dx, dy = dy, -dx
            elif i == 'L': 
                dx, dy = -dy, dx
            elif i == 'G': 
                x, y = x + dx, y + dy
        return (x, y) == (0, 0) or (dx, dy) != (0, 1)

File: src/1041-Robot-Bounded-In-Circle/1041.cpp
class Solution 
{
public:
    bool isRobotBounded(string instructions) 
    {
        int x = 0, y = 0, s = 0;
        int dire[4][2] = {{0, 1}, {-1, 0}, {0, -1}, {1, 0}};
        for (char c : instructions)
        {
            switch (c)
            {
            case 'G':
                x += dire[s][0];
                y += dire[s][1];
                break;
            case 'L':
                s = (s + 1) % 4;
                break;
            case 'R':
                s = (s + 3) % 4;
                break;
            }
        }
        return x == 0 and y == 0 or s > 0;
    }
};